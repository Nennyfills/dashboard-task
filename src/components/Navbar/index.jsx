import PropTypes from 'prop-types';

import { Logo, MenuIcon } from 'assets/svg';
import Loader from 'components/Loader';

const Navbar = ({ loading, firstName, lastName }) => (
  <nav className="navbar" data-testid="navbar">
    <div className="navbar__icon" data-testid="icons">
      <Logo className="navbar__icon--logo" />
      <MenuIcon className="navbar__icon--menu" />
    </div>
    <div className="navbar__avatar">
      <div className="navbar__avatar__holder">
        {loading ? (
          <div className="loading" />
        ) : (
          <div className="navbar__avatar--initials" data-testid="initials">
            <span className="h2">{firstName?.charAt(0)}</span>
            <span className="h2">{lastName?.charAt(0)}</span>
          </div>
        )}
      </div>
      {loading ? (
        <div className="navbar__avatar__loader">
          <div className="single-loader">
            <Loader />
          </div>
        </div>
      ) : (
        <div className="navbar__avatar__details" data-testid="details">
          <p className="bold family roboto">
            {firstName && lastName ? `${firstName} ${lastName}` : 'N/A'}
          </p>
        </div>
      )}
    </div>
  </nav>
);

Navbar.propTypes = {
  firstName: PropTypes.string,
  loading: PropTypes.bool,
  lastName: PropTypes.string,
};

Navbar.defaultProps = {
  firstName: null,
  lastName: null,
  loading: false,
};
export default Navbar;
