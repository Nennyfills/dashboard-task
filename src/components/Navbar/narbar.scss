.navbar {
 max-width: 1750px;
 @include respond(phone) {
  max-width: unset;
 }
 height: 4.5em;
 border-bottom: solid $light-grey 1px;
 @include display-flex(space-between);
 padding: 0.2em 2em;
 @include respond(phone) {
  font-size: 14px;
  padding: 0.2em 0.7em;
 }
 &__icon {
  @include display-flex();
  &--logo {
   margin-right: 2em;
  }
 }
 &__avatar {
  &--initials {
   width: 2.5em;
   height: 2.5em;
   border-radius: 0.7em;
   text-transform: uppercase;
   background: $yellow;
   color: $white;
   @include display-flex(center);
   font-size: 18px;
   @include respond(phone) {
    font-size: 14px;
   }
  }
  @include display-flex(flex-end);

  &__holder {
   margin-right: 0.7em;
   .loading {
    background-color: $light-grey;
   }
  }
  &__details {
   text-transform: capitalize;
   p {
    word-break: break-all;
   }
  }
  &__loader {
   width: 50%;
   margin-left: 2em;
  }
  .font-size {
   @include fontSize(16);
  }
 }
}
