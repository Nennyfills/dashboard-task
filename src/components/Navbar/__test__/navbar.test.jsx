/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable testing-library/prefer-presence-queries */
import { render, screen } from '@testing-library/react';
import Navbar from '..';

it('Should take a snapshot', () => {
  const { asFragment } = render(<Navbar />);
  expect(asFragment(<Navbar />)).toMatchSnapshot();
});

it('Should render a navbar with test id navbar', () => {
  render(<Navbar />);
  const navbarClass = screen.getByTestId('navbar');
  expect(navbarClass).toBeTruthy();
  expect(navbarClass).toHaveClass('navbar');
});

it('Should render a navbar with test id icons', () => {
  render(<Navbar />);
  const navbarClass = screen.getByTestId('icons');
  expect(screen.queryByTestId('icons')).toBeTruthy();
  expect(navbarClass).toHaveClass('navbar__icon');
});

it('Should render a navbar with test id initials', () => {
  render(<Navbar />);
  const navbarClass = screen.getByTestId('initials');
  expect(screen.queryByTestId('initials')).toBeTruthy();
  expect(navbarClass).toHaveClass('navbar__avatar--initials');
});

it('Should render a navbar with test id details', () => {
  render(<Navbar />);
  const navbarClass = screen.getByTestId('details');
  expect(screen.queryByTestId('details')).toBeTruthy();
  expect(navbarClass).toHaveClass('navbar__avatar__details');
});

it('renders a message', () => {
  const { getByText } = render(<Navbar firstName="Joe" lastName="Doe" />);
  expect(getByText('Joe Doe')).toBeInTheDocument();
});
