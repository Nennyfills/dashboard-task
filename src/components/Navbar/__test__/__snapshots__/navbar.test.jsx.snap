// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should take a snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
    data-testid="navbar"
  >
    <div
      class="navbar__icon"
      data-testid="icons"
    >
      <svg
        class="navbar__icon--logo"
      >
        logo.svg
      </svg>
      <svg
        class="navbar__icon--menu"
      >
        menuIcon.svg
      </svg>
    </div>
    <div
      class="navbar__avatar"
    >
      <div
        class="navbar__avatar__holder"
      >
        <div
          class="navbar__avatar--initials"
          data-testid="initials"
        >
          <span
            class="h2"
          />
          <span
            class="h2"
          />
        </div>
      </div>
      <div
        class="navbar__avatar__details"
        data-testid="details"
      >
        <p
          class="bold family roboto"
        >
          N/A
        </p>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
